### PR TITLE
✨ add id on rum events

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

Add id on rum events to better trace them in the backend

## Changes

- Add ids to views, resources, long tasks, errors, actions

## Testing

Unit, local
---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/rawRumEvent.types.ts [hunk 1]
```diff
@@ -112,7 +114,7 @@ export interface RawRumActionEvent {
   date: TimeStamp
   type: RumEventType.ACTION
   action: {
-    id?: string
+    id: string
```
Comment (bcaudan) [inline packages/rum-core/src/rawRumEvent.types.ts:117]: should not the ids be optional?

File: packages/rum-core/src/rumEvent.types.ts [hunk 2]
```diff
@@ -254,7 +254,7 @@ export type RumResourceEvent = CommonProperties & {
     /**
      * UUID of the resource
      */
-    readonly id?: string
+    readonly id: string
```
Comment (bcaudan) [inline packages/rum-core/src/rumEvent.types.ts:257]: should not the id be optional?

Comment (amortemousque) [reply packages/rum-core/src/rumEvent.types.ts]: yes

**Review by bcaudan — APPROVED** [review]
Nice 👍

**Pull request merged**